### ISSUE TITLE
Update minizinc, clean expression

### ISSUE THREAD

**Issue by deliciouslytyped** [NixOS/nixpkgs#104331]
cc @risicle thanks for https://github.com/NixOS/nixpkgs/pull/99034 , it looks like minizinc has now made releases fixing this (?) per https://github.com/MiniZinc/libminizinc/issues/389 , and an update should allow cleaning the patches out of the expression. Ideally an update to 2.5.2 which is the latest release will work, but the fixes should be incorporaed since I would think 2.5.0. (did not check)

**Comment by deliciouslytyped** [comment]
This seems to work for me for minizinc itself, there's some problems with compiling the IDE.
```nix
let 
  inherit (import <nixpkgs> {}) fetchFromGitHub;

  version = "2.5.2";

  overlays = [ (self: super: {
    minizinc = super.minizinc.overrideAttrs (old: {
      inherit version;
      src =  fetchFromGitHub {
        owner = "MiniZinc";
        repo = "libminizinc";
        rev = version;
        sha256 = "1z1la8ckqb8ln26yvi9hs4177lvs4a0l5g90xm2x622mq2f9v1xm";
        };
      patches = [];
      });

    minizincide = super.minizincide.overrideAttrs (old: {
      inherit version;
      src = fetchFromGitHub {
        owner = "MiniZinc";
        repo = "MiniZincIDE";
        rev = version;
        sha256 = "12a781a4fcxmy8l39sr3xvvi0j2gdhnfbggxfggabi2cach84h83";
        };
      });
    }) ];

in
  with import <nixpkgs> { inherit overlays; }; mkShell { buildInputs = [ minizincide gecode ]; } 
```

**Comment by deliciouslytyped** [comment]
Fetching submodules now needs to be enabled due to https://github.com/MiniZinc/MiniZincIDE/issues/116 .
This requires adding `fetchSubmodules = true;` to minizincide.

**Comment by deliciouslytyped** [comment]
The IDE expression should also be updated to use mkDerivation from libsForQt5.callPackage and not stdenv, this will fix wrapper issues with the IDE.

**Comment by jameysharp** [comment]
It looks to me like the problems mentioned in this issue were fixed in #112514. Is that correct, and can this issue be closed?

**Comment by deliciouslytyped** [comment]
If you checked that everything is fine, it's fine by me.
